### PR TITLE
[Enhancement](optimize) optimize for insert_indices_from

### PR DESCRIPTION
**Pull request by BiteTheDDDDt**
# Proposed changes

for tpch Q5:

before:
图片
after:
图片

## Problem summary

Describe your changes.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [ ] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [ ] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [ ] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/vec/columns/column_vector.cpp [hunk 1]
```diff
@@ -324,20 +324,21 @@ void ColumnVector<T>::insert_range_from(const IColumn& src, size_t start, size_t
...
-            // Now Uint8 use to identify null and non null
-            // 1. nullable column : offset == -1 means is null at the here, set true here
-            // 2. real data column : offset == -1 what at is meaningless
-            data[origin_size + i] = (offset == -1) ? T {1} : src_vec.get_element(offset);
-        } else {
-            data[origin_size + i] = (offset == -1) ? T {0} : src_vec.get_element(offset);
+    const T* src_data = reinterpret_cast<const T*>(src.get_raw_data().data);
+
+    if constexpr (std::is_same_v<T, UInt8>) {
+        // nullmap : offset == -1 means is null at the here, set true here
```
Comment (adonis0147) [inline be/src/vec/columns/column_vector.cpp:334]: Is the comment relevant? `offset` is missing here.

File: be/src/vec/columns/column_vector.cpp [hunk 2]
```diff
@@ -324,20 +324,21 @@ void ColumnVector<T>::insert_range_from(const IColumn& src, size_t start, size_t
...
+        for (int i = 0; i < new_size; ++i) {
+            data[origin_size + i] = (indices_begin[i] == -1);
+        }
+    } else {
+        // real data : offset == -1 what at is meaningless
```
Comment (adonis0147) [inline be/src/vec/columns/column_vector.cpp:339]: ditto

File: be/src/vec/columns/column_vector.cpp [hunk 3]
```diff
@@ -349,20 +349,22 @@ void ColumnVector<T>::insert_range_from(const IColumn& src, size_t start, size_t
...
+    if constexpr (std::is_same_v<T, UInt8>) {
+        // nullmap : indices_begin[i] == -1 means is null at the here, set true here
+        for (int i = 0; i < new_size; ++i) {
+            data[origin_size + i] = (indices_begin[i] == -1) +
+                                    (indices_begin[i] != -1) * src_data[indices_begin[i]];
+        }
+    } else {
+        // real data : indices_begin[i] == -1 what at is meaningless
+        for (int i = 0; i < new_size; ++i) {
+            data[origin_size + i] = src_data[indices_begin[i]];
```
Comment (HappenLee) [inline be/src/vec/columns/column_vector.cpp:367]: This is undefined behavior, and might break horribly

**Review by HappenLee — CHANGES_REQUESTED** [review]

**Review by HappenLee — APPROVED** [review]
LGTM

**Pull request merged**